tools/src/main/scala/scala/scalanative/build/IO.scala

```diff
@@ -89,40 +134,79 @@ private[scalanative] object IO {
     val stream       = Files.newInputStream(path)
     val digestStream = new DigestInputStream(stream, digest)
     val buf          = new Array[Byte](bufSize)
-    while (digestStream.read(buf, 0, bufSize) == 1024) {}
+    try {
+      while (digestStream.read(buf, 0, bufSize) != -1) {}
+      digest.digest()
+    } finally {
+      digestStream.close()
+    }
+  }
+
+  /** Compute a SHA-1 hash of `files`. */
+  def sha1files(files: Seq[Path], bufSize: Int = 1024): Array[Byte] = {
+    val digest = MessageDigest.getInstance("SHA-1")
+    val buf    = new Array[Byte](bufSize)
```

Comment:
The `val buf =` should also be inside the `try`.

Reply:
This would create a new buffer for each file rather than just reusing the buffer.